c10/cuda/CUDACachingAllocator.h

```diff
@@ -157,16 +157,19 @@ struct TraceEntry {
       int64_t addr,
       size_t size,
       cudaStream_t stream,
+      int device,
       std::shared_ptr<GatheredContext> context = nullptr)
       : action_(action),
         addr_(addr),
         context_(std::move(context)),
         stream_(stream),
+        device_(device),
         size_(size) {}
   Action action_;
   int64_t addr_; // for OOM, this is the amount of free bytes reported by cuda
   std::shared_ptr<GatheredContext> context_;
   cudaStream_t stream_;
+  int device_;
```

Comment:
Since this is 32-bits, we should place this right after `Action` (also 32-bits), so that the size of struct doesn't get bigger.